application/src/test/java/org/thingsboard/server/controller/AlarmControllerTest.java

```diff
@@ -796,4 +800,101 @@ public void testCreateAlarmWithOtherCustomerAsAssignee() throws Exception {
         doPost("/api/alarm", alarm).andExpect(status().isForbidden());
     }
 
+    @Test
+    public void testGetAlarmTypes() throws Exception {
```

Comment:
You should add the case when two different originators fired the same `alarmType` alarms. This has to test whether the unique constraint is ok and upsert is fine.